src/librekey/g10_sexp.hpp

```diff
@@ -27,92 +27,50 @@
 #ifndef RNP_G10_SEXP_HPP
 #define RNP_G10_SEXP_HPP
 
+#include "sexp/sexp-error.h"
+#include "sexp/sexp.h"
+
 #define SXP_MAX_DEPTH 30
 
-class s_exp_element_t {
-  protected:
-    bool is_block_;
+typedef sexp::sexp_object_t s_exp_element_t;
 
+class s_exp_block_t : public sexp::sexp_string_t {
   public:
-    s_exp_element_t(bool block) : is_block_(block){};
-    virtual ~s_exp_element_t() = default;
-    virtual bool
-    write(pgp_dest_t &dst) const noexcept
+    s_exp_block_t(const uint8_t *bt, size_t ln)
     {
-        return false;
-    };
-
-    bool
-    is_block()
+        data_string = sexp::sexp_simple_string_t(bt, ln);
+    }
+    s_exp_block_t(void) : sexp::sexp_string_t()
+    {
+    }
+    s_exp_block_t(const uint8_t *bt) : sexp::sexp_string_t(bt)
     {
-        return is_block_;
     }
-};
-
-class s_exp_block_t : public s_exp_element_t {
-  protected:
-    std::vector<uint8_t> bytes_;
-
-  public:
-    s_exp_block_t(const uint8_t *bt = NULL, size_t ln = 0)
-        : s_exp_element_t(true),
-          bytes_(bt ? std::vector<uint8_t>(bt, bt + ln) : std::vector<uint8_t>()){};
     s_exp_block_t(unsigned u);
-    s_exp_block_t(const std::string &str)
-        : s_exp_element_t(true),
-          bytes_((uint8_t *) str.data(), (uint8_t *) (str.data() + str.size())){};
-    s_exp_block_t(const pgp_mpi_t &mpi);
-    unsigned as_unsigned() const noexcept;
-    bool     write(pgp_dest_t &dst) const noexcept;
-
-    const std::vector<uint8_t> &
-    bytes() const
+    s_exp_block_t(const std::string &str) : sexp::sexp_string_t(str)
     {
-        return bytes_;
     }
+    s_exp_block_t(const pgp_mpi_t &mpi);
 };
 
-class s_exp_t : public s_exp_element_t {
-    std::vector<std::unique_ptr<s_exp_element_t>> elements_;
+class s_exp_t : public sexp::sexp_list_t {
```

Comment:
It would be better to use some g10-specific C++-style naming, like G10SExp, but that could be done as a separate task.

Reply:
Well, it was the integration point. I left class names and their interfaces unchanged for ```s_exp_t```, ```s_exp_elemement_t``` and put there the new implementation.  I would rather keep it as is within the scope of this specific PR